include/Game/Entities/Kabuto.h

```diff
@@ -2,9 +2,158 @@
+	virtual void startCarcassMotion();                      // _2C4
+	virtual float getDownSmokeScale();                      // _2EC (weak)
+	virtual void doStartMovie();                            // _2F0
+	virtual void doEndMovie();                              // _2F4
+	virtual void setFSM(FSM*);                              // _2F8
+	virtual void createEffect();                            // _2FC (weak)
+	virtual void setupEffect();                             // _300 (weak)
+	virtual void startRotateEffect();                       // _304 (weak)
+	virtual void finishRotateEffect();                      // _308 (weak)
+	virtual void startWaitEffect();                         // _30C (weak)
+	virtual void finishWaitEffect();                        // _310 (weak)
+	virtual void effectDrawOn();                            // _314 (weak)
+	virtual void effectDrawOff();                           // _318 (weak)
+
+	void setRandTarget();
+	void getSearchedTarget();
+	void isAttackableTarget();
+	void createStoneAttack();
+	void updateCaution();
+	void getViewAngle();
+	void lifeIncrement();
+	void createRockEmitEffect();
+
+	// _00		= VTBL
+	// _00-_2BC = EnemyBase
+	FSM* m_kabutoFSM;                    // _2BC
+	WalkSmokeEffect::Mgr m_walkSmokeMgr; // _2C0
+	float _2C8;                          // _2C8
+	int _2CC;                            // _2CC
+	Vector3f m_targetPosition;           // _2D0
+	float _2DC;                          // _2DC
+	u8 _2E0;                             // _2E0, unknown
+	bool m_isUnderground;                // _2E1
+	u8 _2E2[0x2];                        // _2E2, unknown/probably padding
+	                                     // _2E4 = PelletView
+};
+
+struct Mgr : public EnemyMgrBase {
+	Mgr(int, u8);
+
+	virtual ~Mgr() { }                                 // _58 (weak)
+	virtual EnemyTypeID::EEnemyTypeID getEnemyTypeID() // _AC (weak)
+	{
+		return EnemyTypeID::EnemyID_Kabuto;
+	}
+	virtual SysShape::Model* createModel();         // _B0
+	virtual void loadModelData();                   // _C8
+	virtual void loadAnimData();                    // _CC
+	virtual J3DModelData* doLoadBmd(void* resource) // _D4 (weak)
+	{
+		return J3DModelLoaderDataBase::load(resource, 0x01240030);
+	}
+	virtual ResTIMG* getChangeTexture() = 0; // _E0
+
+	// _00		= VTBL
+	// _00-_44	= EnemyMgrBase
+};
+
+struct Parms : public EnemyParmsBase {
+	struct ParmParms : public Parameters {
+		inline ParmParms()
+		    : Parameters(nullptr, "EnemyParmsBase")
+		{
+		}
+	};
+
+	Parms() { }
+
+	virtual void read(Stream& stream) // _08 (weak)
+	{
+		((Parameters*)this)->read(stream);
+		m_general.read(stream);
+		m_kabutoParms.read(stream);
+	}
+
+	// _00-_7F8	= EnemyParmsBase
+	ParmParms m_kabutoParms;
+};
+
+struct FSM : public EnemyStateMachine {
+	virtual void init(EnemyBase*); // _08
+
+	// _00		= VTBL
+	// _00-_1C	= EnemyStateMachine
+};
+
+struct State : public EnemyFSMState {
+	// _00		= VTBL
+	// _00-_10 	= EnemyFSMState
+};
+
+struct StateAttack;
```

Comment:
Don't add newlines for these definitions, keep them one after another:

stateattack
statedead
stateflick

not

stateattack

statedead

stateflick

Reply:
left newlines in here with the hope of finishing these structs before committing, but felt it was enough work for one PR - i've removed the inlines for now and added a comment TODO to fix them up later